feat: add first implementation of collector of sharedmobility

Giacomo92 opened the pull request.

clezag commented on collectors/sharedmobility/.env.example:
Remove all unused parameters, such as PAGING_LIMIT_NAME or RAW_BINARY

clezag commented on collectors/sharedmobility/src/main.go:
```diff
@@ -0,0 +1,154 @@
+			}
+
+			var raw any
+			if env.RAW_BINARY {
```
You are already know it's json. 
Remove the raw functionality, in the rest-poller it's intended for handling non-text mime types, which does not apply here.

Giacomo92 commented:
Hi @clezag I’ve addressed all your points and pushed the transformer directory as well. However, I'm encountering two errors when I test the transformer:

`
app-1  | {"time":"2025-07-24T13:54:00.270630432Z","level":"INFO","msg":"Pushing provenance done","uuid":""}
app-1  | {"time":"2025-07-24T13:54:00.270667041Z","level":"DEBUG","msg":"Syncing data types..."}                                                           
app-1  | {"time":"2025-07-24T13:54:00.274764513Z","level":"ERROR","msg":"Error performing POST"}
app-1  | {"time":"2025-07-24T13:54:00.274809105Z","level":"DEBUG","msg":"Syncing data types done."}                                                        
app-1  | {"time":"2025-07-24T13:54:00.274816744Z","level":"DEBUG","msg":"SyncDataTypes for stationType","stationType":"Scooter"}                           
app-1  | {"time":"2025-07-24T13:54:00.27482866Z","level":"INFO","msg":"Pushing provenance..."}
app-1  | {"time":"2025-07-24T13:54:00.274831452Z","level":"INFO","msg":"prv: 0.1.0 prn: tr-sharedmobility"}                                                
app-1  | {"time":"2025-07-24T13:54:00.279795318Z","level":"ERROR","msg":"Error performing POST"}                                                           
app-1  | {"time":"2025-07-24T13:54:00.279885374Z","level":"ERROR","msg":"error","err":"Post \"https://share.opendatahub.testingmachine.eu/json/provenance?&prn=tr-sharedmobility&prv=0.1.0\": dial tcp: lookup share.opendatahub.testingmachine.eu on 192.168.65.7:53: no such host"}                                 
app-1  | {"time":"2025-07-24T13:54:00.279907835Z","level":"INFO","msg":"Pushing provenance done","uuid":""}
app-1  | {"time":"2025-07-24T13:54:00.279912376Z","level":"DEBUG","msg":"Syncing data types..."}                                                           
app-1  | {"time":"2025-07-24T13:54:00.283645968Z","level":"ERROR","msg":"Error performing POST"}
app-1  | {"time":"2025-07-24T13:54:00.283674093Z","level":"DEBUG","msg":"Syncing data types done."}                                                        
app-1  | {"time":"2025-07-24T13:54:00.283677628Z","level":"INFO","msg":"Data types registration complete"}                                                 
app-1  | {"time":"2025-07-24T13:54:03.309464431Z","level":"ERROR","msg":"failed to initialize Tr pub","err":"failed to dial AMQP: Exception (403) Reason: \"username or password not allowed\""}
app-1  | panic: failed to dial AMQP: Exception (403) Reason: "username or password not allowed"
app-1  |                                                                                                                                                   
app-1  | goroutine 1 [running]:                                                                                                                            
app-1  | github.com/noi-techpark/opendatahub-go-sdk/ingest/tr.NewTr[...]({0xea9da0?, 0x1491300}, {{{0x0, 0x0}}, {0xc00003e067, 0x27}, {0xc00003c03c, 0x6}, {0xc00004404a, 0x11}, ...})                                                                app-1  | github.com/noi-techpark/opendatahub-go-app-1  | github.com/noi-techpark/opendatahub-go-sdk/ingest/tr.NewTr[...]({0xea9da0?, 0x1491300}, {{{0x0, 0x0}}, {0xc00003e067, 0x27}, {0xc00003c03c, 0x6}, {0xc00004404a, 0x11}, ...})
app-1  |        /go/pkg/mod/github.com/noi-techpark/opendatahub-go-sdk/ingest@v1.0.3/tr/tr.go:57 +0x309
app-1  | main.main()
app-1  |        /code/main.go:128 +0xf1
app-1  | exit status 2
app-1 exited with code 1
`

Let me know if you have any suggestions or if there's anything specific you'd like me to adjust.

Thanks!
